card/mod/01_core/set/all/permissions.rb

```diff
@@ -138,24 +145,26 @@ def ok_to_comment
   deny_because 'No comments allowed on structured content' if structure
 end
 
-event :set_read_rule, before: :store do
-  if trash == true
-    self.read_rule_id = self.read_rule_class = nil
-  else
-    # avoid doing this on simple content saves?
-    rcard, rclass = permission_rule_card(:read)
-    self.read_rule_id = rcard.id
-    self.read_rule_class = rclass
-    # find all cards with me as trunk and update their read_rule
-    # (because of *type plus right)
-    # skip if name is updated because will already be resaved
-
-    if !new_card? && type_id_changed?
-      Auth.as_bot do
-        Card.search(left: self.name).each do |plus_card|
-          plus_card = plus_card.refresh.update_read_rule
-        end
-      end
+event :clear_read_rule, before: :store, on: :delete do
+  self.read_rule_id = self.read_rule_class = nil
+end
+
+event :set_read_rule, before: :store, on: :save do
+  # avoid doing this on simple content saves?
+  read_rule_id, read_rule_class = permission_rule_id_and_class(:read)
+  self.read_rule_id = read_rule_id
+  self.read_rule_class = read_rule_class
```

Comment:
I did some cleanup here because this was slow, but the real reason was the unnecessary cache clearing